Mathlib/Topology/Metrizable/Basic.lean

```diff
@@ -5,129 +5,165 @@ Authors: Yury Kudryashov
-  letI : PseudoMetricSpace X := pseudoMetrizableSpacePseudoMetric X
-  letI : PseudoMetricSpace Y := pseudoMetrizableSpacePseudoMetric Y
+  let : UniformSpace X := pseudoMetrizableSpaceUniformity X
+  have : (uniformity X).IsCountablyGenerated :=
+    pseudoMetrizableSpaceUniformity_countably_generated X
+  let : UniformSpace Y := pseudoMetrizableSpaceUniformity Y
+  have : (uniformity Y).IsCountablyGenerated :=
+    pseudoMetrizableSpaceUniformity_countably_generated Y
   inferInstance
 
 /-- Given an inducing map of a topological space into a pseudo metrizable space, the source space
 is also pseudo metrizable. -/
 theorem _root_.Topology.IsInducing.pseudoMetrizableSpace [PseudoMetrizableSpace Y] {f : X → Y}
     (hf : IsInducing f) : PseudoMetrizableSpace X :=
-  letI : PseudoMetricSpace Y := pseudoMetrizableSpacePseudoMetric Y
-  ⟨⟨hf.comapPseudoMetricSpace, rfl⟩⟩
+  let u : UniformSpace Y := pseudoMetrizableSpaceUniformity Y
+  have : (uniformity Y).IsCountablyGenerated :=
+    pseudoMetrizableSpaceUniformity_countably_generated Y
+  ⟨⟨u.comap f, u.toTopologicalSpace_comap.trans hf.eq_induced.symm,
+    Filter.comap.isCountablyGenerated (uniformity Y) (Prod.map f f)⟩⟩
 
 /-- Every pseudo-metrizable space is first countable. -/
 instance (priority := 100) PseudoMetrizableSpace.firstCountableTopology
-    [h : PseudoMetrizableSpace X] : FirstCountableTopology X := by
-  rcases h with ⟨_, hm⟩
-  rw [← hm]
-  exact @UniformSpace.firstCountableTopology X PseudoMetricSpace.toUniformSpace
-    EMetric.instIsCountablyGeneratedUniformity
+    [h : PseudoMetrizableSpace X] : FirstCountableTopology X :=
+  let : UniformSpace X := pseudoMetrizableSpaceUniformity X
+  have : (uniformity X).IsCountablyGenerated :=
+    pseudoMetrizableSpaceUniformity_countably_generated X
+  inferInstance
 
 instance PseudoMetrizableSpace.subtype [PseudoMetrizableSpace X] (s : Set X) :
     PseudoMetrizableSpace s :=
   IsInducing.subtypeVal.pseudoMetrizableSpace
 
 instance pseudoMetrizableSpace_pi [∀ i, PseudoMetrizableSpace (A i)] :
-    PseudoMetrizableSpace (∀ i, A i) := by
-  cases nonempty_fintype ι
-  letI := fun i => pseudoMetrizableSpacePseudoMetric (A i)
-  infer_instance
+    PseudoMetrizableSpace (∀ i, A i) :=
+  let := fun i => pseudoMetrizableSpaceUniformity (A i)
+  have := fun i => pseudoMetrizableSpaceUniformity_countably_generated (A i)
+  inferInstance
+
+instance PseudoMetrizableSpace.regularSpace [PseudoMetrizableSpace X] : RegularSpace X :=
+  let := pseudoMetrizableSpaceUniformity X
+  inferInstance
 
 /-- A topological space is metrizable if there exists a metric space structure compatible with the
-topology. To endow such a space with a compatible distance, use
+topology. To endow such a space with a compatible uniformity, use
+`letI : UniformSpace X := TopologicalSpace.pseudoMetrizableSpaceUniformity X`.
+To endow such a space with a compatible distance, use
 `letI : MetricSpace X := TopologicalSpace.metrizableSpaceMetric X`. -/
```

Comment:
This is subject to the same / similar comments I had for `PseudoMetrizableSpace`.